src/contexts/TaskDataProvider.js

```diff
@@ -72,10 +73,20 @@ const TaskDataProvider = ({ children }) => {
       setExistingTask(resolvedCreateTask.task);
       // every 5 seconds, ping tasks endpoint to update status of task
       // this status is used by the ExportData component for conditional rendering
-      setInterval(async () => {
+      // clear and exit setInterval if task status is 'completed' or MAX_TASK_QUERIES are made
```

Comment:
Nice solution. Two suggestions:

1. Use `taskQueriesMade >= MAX_TASK_QUERIES` just in case for some weird reason it skips # `100`.
2. Not for this commit, but maybe a next one, you should have a "timeout" state where you show like a clock icon (Material Icons / Update) and a tooltip like: "This task is taking longer than expected. Please refresh the page or click here to continue waiting for it to finish." And if you click on it, it resets the counter to 0.

Reply:
👍 created an issue for this https://github.com/graasp/graasp-research/issues/51